lib/services/s3.js

```diff
@@ -479,6 +499,49 @@ AWS.util.update(AWS.S3.prototype, {
     }
   },
 
+  /**
+   * @api private
+   */
+  validateMd5TrailingChecksum: function validateMd5TrailingChecksum(resp) {
+    var response = resp.httpResponse;
+    var buffer = response.body;
```

Comment:
In node.js, the response body is likely to be a stream. It might be easier in node to create a passthrough stream that generates the md5 hash and compares against the checksum. You'll know the expected content-length since it's passed in the response headers. Your solution here should still cover the browser use-case though.

Reply:
Mark: This part will only be performed on Browser.